contribs/src/main/java/com/netflix/conductor/contribs/http/HttpTask.java

```diff
@@ -80,8 +86,21 @@ public void start(Workflow workflow, Task task, WorkflowExecutor executor) throw
 			task.setStatus(Status.FAILED);
 			return;
 		} else {
-			if (url != null) {
-				input.setUri(url + input.getUri());
+			final String uri = input.getUri();
+			url = StringUtils.defaultIfEmpty(url, "");
```

Comment:
This is redundant as above 'input.getUri() == null' checked for null and you can safely use next 'if (uri.startsWith("/"))'

Reply:
The `url` variable is only set if a service discovery query is provided. In the case that a query is not provided, but a URI is, then `url` will still be `null`. We have `url` and `uri` variables here, maybe it makes sense to rename `url` to eliminate some of the confusion since they look so similar.

Reply:
Renamed in a1b00076.